ksql-metastore/src/main/java/io/confluent/ksql/metastore/MetaStoreImpl.java

```diff
@@ -112,18 +137,84 @@ public void deleteSource(final String sourceName) {
   @Override
   public void putAll(MetaStore otherMetaStore) {
     this.topicMap.putAll(otherMetaStore.getAllKsqlTopics());
-    this.dataSourceMap.putAll(otherMetaStore.getAllStructuredDataSources());
+    this.dataSourceMap.putAll(otherMetaStore.getDataSourceMap());
+  }
+
+  @Override
+  public void addSourceNames(Set<String> sourceNames, String queryId) {
+    for (String sourceName: sourceNames) {
```

Comment:
can we just delegate this to `ReferentialIntegrityTable`? i.e, so we don't expose the set via `getSourceForQueries`?

Same for `addSinkNames`

Reply:
Removed both of them.